Feature request: bundle wal2json

Great app, thanks for making PostgreSQL so easy to install!

I'd like to request that you bundle the `wal2json` output plugin with PostgresApp. It only takes a couple seconds to build, but unfortunately because PostgresApp is built with a specific version of the MacSDKs, users who do not have that SDK installed cannot build against it. (Note many cloud providers such as [Amazon RDS bundle wal2json](https://aws.amazon.com/about-aws/whats-new/2018/11/amazon-rds-postgresql-105-govcloud/).)

For example, the `pg_config` in this user's Postgres.app indicates `/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk`, but this user only has `/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk` available:

```
+ rm -Rf wal2json
+ git clone https://github.com/eulerto/wal2json.git
Cloning into 'wal2json'...
remote: Enumerating objects: 43, done.
remote: Counting objects: 100% (43/43), done.
remote: Compressing objects: 100% (24/24), done.
remote: Total 492 (delta 24), reused 29 (delta 19), pack-reused 449
Receiving objects: 100% (492/492), 203.11 KiB | 3.44 MiB/s, done.
Resolving deltas: 100% (343/343), done.
+ cd wal2json
+ USE_PGXS=1
+ make
gcc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -Wno-unused-command-line-argument  -mmacosx-version-min=10.7  -I. -I./ -I/Applications/Postgres.app/Contents/Versions/10/include/postgresql/server -I/Applications/Postgres.app/Contents/Versions/10/include/postgresql/internal -I/Applications/Postgres.app/Contents/Versions/10/share/icu -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk  -I/Applications/Postgres.app/Contents/Versions/10/include/libxml2  -I/Applications/Postgres.app/Contents/Versions/10/include  -c -o wal2json.o wal2json.c
clang: warning: no such sysroot directory: '/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk' [-Wmissing-sysroot]
In file included from wal2json.c:13:
In file included from /Applications/Postgres.app/Contents/Versions/10/include/postgresql/server/postgres.h:47:
/Applications/Postgres.app/Contents/Versions/10/include/postgresql/server/c.h:81:10: fatal error: 'stdio.h' file not found
#include <stdio.h>
         ^~~~~~~~~
1 error generated.
make: *** [wal2json.o] Error 1
```

```
$ ls /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer//SDKs/
MacOSX.sdk      MacOSX10.14.sdk
```

Installing wal2json is straight-forward (and fast!):

```bash
PATH="/path/to/pg/bin:$PATH"
git clone https://github.com/eulerto/wal2json.git
cd wal2json
USE_PGXS=1 make
USE_PGXS=1 make install
```

--- a/README.md
+++ b/README.md
@@ -26,6 +26,7 @@ Documentation is available at [http://postgresapp.com/documentation](http://post
 - [PostgreSQL](http://www.postgresql.org/)
 - [PostGIS](http://postgis.net/)
 - [plv8](http://code.google.com/p/plv8js/wiki/PLV8)
+- [wal2json](https://github.com/eulerto/wal2json)
 
 ## How To Build
 
@@ -47,7 +48,7 @@ For building PostgreSQL with docs, you also need a bunch of other tools:
 Then make sure you remove other versions of `Postgres.app` from your Applications folder.
 
 Open the `src` directory and type `make`.
-This will download and build PostgreSQL, PostGIS, and PLV8. 
+This will download and build PostgreSQL, PostGIS, PLV8, and wal2json.
 Several hundred megabytes of sources will be downloaded and built.
 This can take an hour or longer, depending on your Internet connection and processor speed.
 All the products will be installed in `/Applications/Postgres.app/Contents/MacOS/`.

--- a/src-10/makefile
+++ b/src-10/makefile
@@ -42,6 +42,9 @@ PLV8_VERSION=1.4.8
 # https://s3.amazonaws.com/json-c_releases/releases/index.html
 JSONC_VERSION=0.12.1
 
+# https://github.com/eulerto/wal2json
+WAL2JSON_VERSION=wal2json_1_0
+
 # http://simplesystems.org/libtiff/
 LIBTIFF_VERSION=4.0.10
 
@@ -71,7 +74,7 @@ CURL=/usr/bin/curl -L10 --silent --show-error --remote-name
 TAR=/usr/bin/tar xzf
 
 all: postgresapp
-clean: clean-postgresql clean-openssl clean-libxml2 clean-libgdal clean-libproj clean-postgis clean-plv8 clean-libjpeg clean-libjasper clean-libgeos clean-libtiff clean-json-c
+clean: clean-postgresql clean-openssl clean-libxml2 clean-libgdal clean-libproj clean-postgis clean-plv8 clean-libjpeg clean-libjasper clean-libgeos clean-libtiff clean-json-c clean-wal2json
 	rm -Rf "$(PREFIX)"
 check: check-postgresql
 
@@ -105,7 +108,7 @@ $(EXPORT_ARCHIVE_PATH)/Postgres.app: $(EXPORT_ARCHIVE_PATH)/Postgres.xcarchive
 	xcodebuild -exportArchive -archivePath $(EXPORT_ARCHIVE_PATH)/Postgres.xcarchive -exportPath $(EXPORT_ARCHIVE_PATH) -exportOptionsPlist "$(CURDIR)"/exportOptions.plist POSTGRESAPP_VERSION=$(POSTGRESAPP_VERSION) POSTGRES_MAJOR_VERSION=$(POSTGRES_MAJOR_VERSION)
 	touch $@
 
-$(EXPORT_ARCHIVE_PATH)/Postgres.xcarchive: postgresql postgis plv8
+$(EXPORT_ARCHIVE_PATH)/Postgres.xcarchive: postgresql postgis plv8 wal2json
 	xcodebuild archive -project "$(XCODE_PROJECT_FILE)" -scheme Postgres -archivePath $(EXPORT_ARCHIVE_PATH)/Postgres.xcarchive POSTGRESAPP_VERSION=$(POSTGRESAPP_VERSION) POSTGRES_MAJOR_VERSION=$(POSTGRES_MAJOR_VERSION)
 
 
@@ -504,3 +507,24 @@ v8-${V8_VERSION}.tar.gz:
 
 clean-plv8:
 	rm -Rf plv8js v8-${V8_VERSION} plv8-${PLV8_VERSION}
+
+
+#########################
+###### wal2json #########
+#########################
+
+wal2json: $(PREFIX)/lib/postgresql/wal2json.so
+	touch $@
+
+$(PREFIX)/lib/postgresql/wal2json.so: wal2json-${WAL2JSON_VERSION}/Makefile $(PREFIX)/bin/psql
+	export PGHOME="$(PREFIX)"; export PG_CONFIG="$(PREFIX)/bin/pg_config"; export USE_PGXS=1; make -C wal2json-${WAL2JSON_VERSION} install
+
+wal2json-${WAL2JSON_VERSION}/Makefile: wal2json-${WAL2JSON_VERSION}.tar.gz
+	$(TAR) $<
+	touch $@
+
+wal2json-${WAL2JSON_VERSION}.tar.gz:
+	/usr/bin/curl -L10 --silent --fail --show-error -o $@ https://github.com/eulerto/wal2json/archive/${WAL2JSON_VERSION}.tar.gz
+
+clean-wal2json:
+	rm -Rf wal2json-${WAL2JSON_VERSION}

--- a/src-11/makefile
+++ b/src-11/makefile
@@ -38,6 +38,9 @@ PLV8_VERSION=2.3.12
 # https://s3.amazonaws.com/json-c_releases/releases/index.html
 JSONC_VERSION=0.13.1
 
+# https://github.com/eulerto/wal2json
+WAL2JSON_VERSION=wal2json_1_0
+
 # https://github.com/protocolbuffers/protobuf/releases
 PROTOBUF_VERSION=3.9.1
 
@@ -78,8 +81,8 @@ export PROTOC protobuf_CFLAGS protobuf_LIBS
 CURL=/usr/bin/curl -L10 --silent --show-error --remote-name --fail
 TAR=/usr/bin/tar xzf
 
-all: postgresql postgis plv8
-clean: clean-postgresql clean-openssl clean-libxml2 clean-libgdal clean-libproj clean-postgis clean-plv8 clean-libjpeg clean-libjasper clean-libgeos clean-libtiff clean-json-c clean-icu clean-protobuf-c clean-protobuf-cpp
+all: postgresql postgis plv8 wal2json
+clean: clean-postgresql clean-openssl clean-libxml2 clean-libgdal clean-libproj clean-postgis clean-plv8 clean-libjpeg clean-libjasper clean-libgeos clean-libtiff clean-json-c clean-icu clean-protobuf-c clean-protobuf-cpp clean-wal2json
 	rm -Rf "$(PREFIX)"
 check: check-postgresql
 
@@ -500,3 +503,24 @@ plv8-${PLV8_VERSION}.tar.gz:
 
 clean-plv8:
 	rm -Rf plv8-${PLV8_VERSION}
+
+
+#########################
+###### wal2json #########
+#########################
+
+wal2json: $(PREFIX)/lib/postgresql/wal2json.so
+	touch $@
+
+$(PREFIX)/lib/postgresql/wal2json.so: wal2json-${WAL2JSON_VERSION}/Makefile $(PREFIX)/bin/psql
+	export PGHOME="$(PREFIX)"; export PG_CONFIG="$(PREFIX)/bin/pg_config"; export USE_PGXS=1; make -C wal2json-${WAL2JSON_VERSION} install
+
+wal2json-${WAL2JSON_VERSION}/Makefile: wal2json-${WAL2JSON_VERSION}.tar.gz
+	$(TAR) $<
+	touch $@
+
+wal2json-${WAL2JSON_VERSION}.tar.gz:
+	/usr/bin/curl -L10 --silent --fail --show-error -o $@ https://github.com/eulerto/wal2json/archive/${WAL2JSON_VERSION}.tar.gz
+
+clean-wal2json:
+	rm -Rf wal2json-${WAL2JSON_VERSION}

--- a/src-12/makefile
+++ b/src-12/makefile
@@ -38,6 +38,9 @@ PLV8_VERSION=2.3.13
 # https://s3.amazonaws.com/json-c_releases/releases/index.html
 JSONC_VERSION=0.13.1
 
+# https://github.com/eulerto/wal2json
+WAL2JSON_VERSION=wal2json_1_0
+
 # https://github.com/protocolbuffers/protobuf/releases
 PROTOBUF_VERSION=3.9.2
 
@@ -78,8 +81,8 @@ export PROTOC protobuf_CFLAGS protobuf_LIBS
 CURL=/usr/bin/curl -L10 --silent --show-error --remote-name --fail
 TAR=/usr/bin/tar xzf
 
-all: postgresql postgis plv8
-clean: clean-postgresql clean-openssl clean-libxml2 clean-libgdal clean-libproj clean-postgis clean-plv8 clean-libjpeg clean-libjasper clean-libgeos clean-libtiff clean-json-c clean-icu clean-protobuf-c clean-protobuf-cpp
+all: postgresql postgis plv8 wal2json
+clean: clean-postgresql clean-openssl clean-libxml2 clean-libgdal clean-libproj clean-postgis clean-plv8 clean-libjpeg clean-libjasper clean-libgeos clean-libtiff clean-json-c clean-icu clean-protobuf-c clean-protobuf-cpp clean-wal2json
 	rm -Rf "$(PREFIX)"
 check: check-postgresql
 
@@ -501,3 +504,24 @@ plv8-${PLV8_VERSION}.tar.gz:
 
 clean-plv8:
 	rm -Rf plv8-${PLV8_VERSION}
+
+
+#########################
+###### wal2json #########
+#########################
+
+wal2json: $(PREFIX)/lib/postgresql/wal2json.so
+	touch $@
+
+$(PREFIX)/lib/postgresql/wal2json.so: wal2json-${WAL2JSON_VERSION}/Makefile $(PREFIX)/bin/psql
+	export PGHOME="$(PREFIX)"; export PG_CONFIG="$(PREFIX)/bin/pg_config"; export USE_PGXS=1; make -C wal2json-${WAL2JSON_VERSION} install
+
+wal2json-${WAL2JSON_VERSION}/Makefile: wal2json-${WAL2JSON_VERSION}.tar.gz
+	$(TAR) $<
+	touch $@
+
+wal2json-${WAL2JSON_VERSION}.tar.gz:
+	/usr/bin/curl -L10 --silent --fail --show-error -o $@ https://github.com/eulerto/wal2json/archive/${WAL2JSON_VERSION}.tar.gz
+
+clean-wal2json:
+	rm -Rf wal2json-${WAL2JSON_VERSION}

